plugins/woocommerce/src/Internal/Utilities/COTMigrationUtil.php

```diff
@@ -96,7 +96,7 @@ public function get_post_or_object_meta( ?WP_Post $post, ?\WC_Data $data, string
 			}
 			return $data->get_meta( $key, $single );
 		} else {
-			return get_post_meta( $post->ID, $key, $single );
+			return isset( $post->ID ) ? get_post_meta( $post->ID, $key, $single ) : false;
```

Comment:
Just noting that I've decided to use `false` as "default" value when `$post->ID` is not set, just to make `get_post_or_object_meta ()` mimic what `get_post_meta()` does: return `false` for invalid `$post_id` ([see codex](http://developer.wordpress.org/reference/functions/get_post_meta/)). This, however, makes probably little difference in real use.